gstd/gstd_parser.c

```diff
@@ -938,3 +951,210 @@ gstd_parser_signal_timeout (GstdSession * session, gchar * action, gchar * args,
 
   return ret;
 }
+
+static GstdReturnCode
+gstd_parser_pipeline_create_ref (GstdSession * session, gchar * action,
+    gchar * args, gchar ** response)
+{
+  gchar **tokens;
+  gchar *description;
+  gchar *uri;
+  GstdObject *pipeline_list_node;
+  GstdObject *pipeline_node;
+  GstdReturnCode ret;
+
+  g_return_val_if_fail (GSTD_IS_SESSION (session), GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (action, GSTD_NULL_ARGUMENT);
+  g_return_val_if_fail (args, GSTD_NULL_ARGUMENT);
+
+  tokens = g_strsplit (args, " ", 2);
+
+  /* Get the pipeline list node */
+  ret = gstd_get_by_uri (session, "/pipelines", &pipeline_list_node);
+  if (ret) {
+    goto pipeline_list_node_error;
+  }
+
+  GST_OBJECT_LOCK (session);
+
+  /* Look for the pipeline node */
+  pipeline_node =
+      gstd_list_find_child (GSTD_LIST (pipeline_list_node), tokens[0]);
+
+  /* Pipeline doesn't exist */
+  if (!pipeline_node) {
+    uri = g_strdup_printf ("/pipelines %s", args ? args : "");
+    ret =
+        gstd_parser_parse_raw_cmd (session, (gchar *) "create", uri, response);
+    g_free (uri);
+    if (ret) {
+      goto create_error;
+    }
+    pipeline_node =
+        gstd_list_find_child (GSTD_LIST (pipeline_list_node), tokens[0]);
+  } else {
```

Comment:
What is the response in the case of the pipeline already existing?

Reply:
"null"... Same as the normal pipeline create:
```
gstd> pipeline_create p0 videotestsrc ! fakesink
{
  "code" : 8,
  "description" : "Resource already exists",
  "response" : null
}
```